### PR TITLE
turn the watch cache on

### PR DESCRIPTION
**Pull request by p0lyn0mial**
## Summary
requires https://github.com/kcp-dev/kubernetes/pull/73

## Related issue(s)

Fixes #

### CODE REVIEW

File: test/e2e/watchcache/watchcache_enabled_test.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kubernetesclientset "k8s.io/client-go/kubernetes"
+
+	"github.com/kcp-dev/kcp/test/e2e/fixtures/wildwest"
+	wildwestv1alpha1 "github.com/kcp-dev/kcp/test/e2e/fixtures/wildwest/apis/wildwest/v1alpha1"
+	wildwestclientset "github.com/kcp-dev/kcp/test/e2e/fixtures/wildwest/client/clientset/versioned"
+	"github.com/kcp-dev/kcp/test/e2e/framework"
+)
+
+func TestWatchCacheEnabled(t *testing.T) {
```
Comment (sttts) [inline test/e2e/watchcache/watchcache_enabled_test.go:39]: use extensive `t.Logf` steps in the shape of a story line.

File: test/e2e/watchcache/watchcache_enabled_test.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+	server := framework.SharedKcpServer(t)
+	ctx, cancel := context.WithCancel(context.Background())
+	t.Cleanup(cancel)
+	org := framework.NewOrganizationFixture(t, server)
+	cluster := framework.NewWorkspaceFixture(t, server, org)
+	cfg := server.DefaultConfig(t)
+	kubeClusterClient, err := kubernetesclientset.NewClusterForConfig(cfg)
+	require.NoError(t, err)
+
+	// create cowboys.wildwest.dev CRD and an object
```
Comment (sttts) [inline test/e2e/watchcache/watchcache_enabled_test.go:50]: can we have the same for APIBindings and native resources?

File: test/e2e/watchcache/watchcache_enabled_test.go [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+	rsp := kubeClusterClient.RESTClient().Get().AbsPath("/metrics").Do(ctx)
+	raw, err := rsp.Raw()
+	require.NoError(t, err)
+	scanner := bufio.NewScanner(strings.NewReader(string(raw)))
+	var totalCacheHits, cowboysCacheHit int
+	for scanner.Scan() {
+		txt := scanner.Text()
+		if strings.Contains(txt, "apiserver_cache_list_total") {
+			totalCacheHits++
+			if strings.Contains(txt, `resource_prefix="/wildwest.dev/cowboys/customresources`) {
```
Comment (sttts) [inline test/e2e/watchcache/watchcache_enabled_test.go:84]: oh, this means we need work in metrics. It does not distinguish between workspaces and not between CRDs and APIBindings, etc.

Comment (sttts) [reply test/e2e/watchcache/watchcache_enabled_test.go:84]: Also note that this test is run in parallel with other e2e also using the cowboys. Can we use a custom API group here, e.g. with a uuid as group name?

Comment (p0lyn0mial) [reply test/e2e/watchcache/watchcache_enabled_test.go:84]: I think we should be able to distinguish by 
```
resource_prefix="/group/resource/idenity"}
resource_prefix="/group/resource/customresources"}
```

we could pick up a different `gr`

File: test/e2e/watchcache/watchcache_enabled_test.go (hunk 2 above)
Comment (p0lyn0mial) [reply test/e2e/watchcache/watchcache_enabled_test.go:50]: yes, for native we could check ```/core/secrets```

File: test/e2e/watchcache/watchcache_enabled_test.go (hunk 3 above)
Comment (p0lyn0mial) [reply test/e2e/watchcache/watchcache_enabled_test.go:84]: does it make sense?

Comment (sttts) [reply test/e2e/watchcache/watchcache_enabled_test.go:84]: actually, are ok if some other controller increases the metrics? Then we shouldn't care about unique GRs.

File: test/e2e/watchcache/watchcache_enabled_test.go (hunk 1 above)
Comment (sttts) [reply test/e2e/watchcache/watchcache_enabled_test.go:39]: still open.

**Pull request merged**